web/src/main/java/com/bazaarvoice/emodb/web/compactioncontrol/CompactionControlMonitor.java

```diff
@@ -44,19 +49,32 @@ protected void shutDown() {
     @Override
     protected void runOneIteration() {
         try {
+            updateMetrics(_clock.millis());
             deleteExpiredStashTimes(_clock.millis());
         } catch (Throwable t) {
             _log.error("Unexpected exception.", t);
         }
-        ;
+    }
+
+    private void updateMetrics(long currentTimeInMillis) {
+        Map<String, StashRunTimeInfo> stashTimeInfoMap = _compactionControlSource.getAllStashTimes();
+        if (stashTimeInfoMap.size() > 0) {
+            DateTime oldestCompactionControlTime = new DateTime(stashTimeInfoMap.entrySet().stream()
+                    .min((entry1, entry2) -> entry1.getValue().getTimestamp() > entry2.getValue().getTimestamp() ? 1 : -1)
+                    .get().getValue().getTimestamp());
+            long lag = currentTimeInMillis - oldestCompactionControlTime.getMillis();
+            _metricRegistry.register(MetricRegistry.name("bv.emodb.scan", "ScanUploader", "compaction-control-time-lag"), (Gauge<Long>) () -> lag);
```

Comment:
This will actually cause an exception because it's trying to replace a metric which already exists.  A better approach would be for `updateMetrics()` to write the lag to a volatile class attribute, such as `_lag`, and then register the metric once with a Gauge which simply returns `_lag`.